stdlib/builtin.rbi

```diff
@@ -560,10 +560,60 @@ class Enumerator<'a, 'b>
                 | -> Enumerator<['a, Integer], 'b>
 end
 
+class Encoding
+end
+
 class Regexp
   def self.compile: (String, *any) -> Regexp
+                  | (Regexp) -> Regexp
   def self.escape: (String) -> String
+  def self.last_match: -> MatchData?
+                     | (Integer) -> String?
+  def self.quote: (String) -> String
+  def self.try_convert: (any) -> Regexp?
+  def self.union: (*String) -> Regexp
+                | (Array<String>) -> Regexp
+                | (*Regexp) -> Regexp
+                | (Array<Regexp>) -> Regexp
+
+  def initialize: (String, *any) -> any
+                  | (Regexp) -> any
+  def ===: (String) -> bool
+  def =~: (String) -> Integer?
+  def casefold?: -> bool
+  def encoding: -> Encoding
+  def fixed_encoding?: -> bool
+  def match: (String) -> MatchData?
+           | (String, Integer) -> MatchData?
+           | (String) { (MatchData) -> void } -> void
```

Comment:
`<'a> (String, Integer?) { (MatchData) -> 'a } -> ('a | nil)`

Reply:
Good! It works well. 😄 

I have one question. If changing the following,

```diff
   def match: (String) -> MatchData?
            | (String, Integer) -> MatchData?
-           | <'a> (String) { (MatchData) -> 'a } -> ('a | nil)
-           | <'a> (String, Integer) { (MatchData) -> 'a } -> ('a | nil)
+           | <'a> (String, Integer?) { (MatchData) -> 'a } -> ('a | nil)
```

some errors are raised. 😕 

```
$ bin/smoke_runner.rb smoke/regexp
🏇 Running smoke test in smoke/regexp...
  🤦‍♀️ Unexpected error found: smoke/regexp/b.rb:1:0: UnexpectedBlockGiven: method_type=(String) -> MatchData? (/(?<foo>a)/.match("a") do |match|)
  🤦‍♀️ Unexpected error found: smoke/regexp/a.rb:68:0: UnexpectedBlockGiven: method_type=(String) -> MatchData? (/a/.match("a") { |m| match_3 = m })
```

Can we make method override more simple?